client/verifier.go

```diff
@@ -46,17 +47,21 @@ func (c *logVerifier) VerifyRoot(trusted, newRoot *trillian.SignedLogRoot,
 	consistency [][]byte) error {
 
 	// Verify SignedLogRoot signature.
-	hash := tcrypto.HashLogRoot(*newRoot)
-	if err := tcrypto.Verify(c.pubKey, hash, newRoot.Signature); err != nil {
-		return err
+	if (newRoot == nil){
+		return fmt.Errorf("VerifyRoot() error: newRoot == nil.")
+	}else {
```

Comment:
Not Gary, but it looks good. :)

Just clarifying, it's somewhat of a general Go-style rule that, if you return inside the if, you omit the else clause.

Eg, we prefer:

```go
if something {
  return fmt.Errorf("...")
}
// code continues here
```

instead of

```go
if something {
  return fmt.Errorf("...")
} else {
  // code continues here
}
```